### DISCUSSION TITLE
Any kotlin/java implementation of authentication?

### DISCUSSION THREAD

**Discussion by iqtedar123** [timdorr/tesla-api#429] [Q&A]
Has anyone created a kotlin/java implementation of the new auth flow? Including MFA support?
Thanks.

**Comment by tbee** [comment]
I did 
https://github.com/tbee/teslaApi

**Comment by tbee** [comment]
But... 
https://www.tbee.org/2021/09/11/i-need-help/

So I'm trying to see if this Ruby implementation also returns a 500. But it seems the version with MFA is not in the Rubygems repo...

**Reply by iqtedar123** [reply]
Does your project work now?
Thanks.

**Reply by tbee** [reply]
No. I gave up (https://teslatasks.com/). Tesla kept changing the login process, but the captcha was the final drop. I've tried creating a website, JavaFX, and Android application to get that to work, but in the end this is just a hobby project and not worth the time and frustration it costs me.

**Reply by iqtedar123** [reply]
Thanks for the update. Sad to hear that.
If you want to get similar functionality to Tesla tasks, you can use home assistant.
Install HA, and the Tesla plugin.
Add Google calendar and create automations that are triggered based off events.

**Reply by tbee** [reply]
Yes, that is possible now. I've written TeslaTasks four years ago, and since it is a service dedicated to this, it probably is easier to use than custom automations.  But I"ll take a peek. Thanks for suggesting.

**Reply by tbee** [reply]
I see there are refresh token generators now (one of the functions TeslaTask also had). That could be a new way of running it again.
